ostd/src/arch/x86/kernel/apic/ioapic.rs

```diff
@@ -197,6 +196,15 @@ pub fn init() {
             let mut vec = Vec::new();
             for id in 0..apic.io_apics.len() {
                 let io_apic = apic.io_apics.get(id).unwrap();
+                #[cfg(feature = "cvm_guest")]
+                // SAFETY:
+                // This is safe because we are ensuring that the `io_apic.address` is a valid MMIO address before this operation.
+                // We are also ensuring that we are only unprotecting a single page.
+                if tdx_is_enabled() {
+                    unsafe {
+                        tdx_guest::unprotect_gpa_range(io_apic.address as usize, 1).unwrap();
```

Comment:
Is `io_apic.address` always page aligned? And is the entire page of `io_apic.address..io_apic.address + PAGE_SIZE` part to IOAPIC or secure to be shared?

Reply:
Yes, it is typically page-aligned in x86. When the space occupied by the I/O APIC is less than a page, the remaining space in that page is typically not used.